src/main/java/org/dita/dost/util/FileUtils.java

```diff
@@ -615,5 +615,38 @@ public static boolean directoryContains(final File directory, final File child)
             return c.getPath().startsWith(d.getPath());
         }
     }
+    
+    /**
+     * Get a file path.
+     * @param parent The parent file.
+     * @param path An URI containing a path which can be either absolute or relative. 
+     * @return The file path, never <code>null</code>
+     */
+    public static File getFilePath(File parent, URI path) {
+        return getFilePath(parent, path.getPath());
+    }
+    
+    /**
+     * Get a file path.
+     * @param parent The parent file.
+     * @param path A file containing a path which can be either absolute or relative. 
+     * @return The file path, never <code>null</code>
+     */
+    public static File getFilePath(File parent, File path) {
+        return getFilePath(parent, path.getPath());
+    }
 
+    /**
+     * Get a file path.
+     * @param parent The parent file.
+     * @param path A file path which can be either absolute or relative. 
+     * @return The file path, never <code>null</code>
+     */
+    public static File getFilePath(File parent, String path) {
+        File filePath = new File(path);
+        if(!filePath.isAbsolute()) {
+            filePath = new File(parent, path);
+        }
+        return filePath;
```

Comment:
If we change this to

```suggestion
        return parent.toPath().resolve(Paths.get(path)).toFile();
```

It would be an easy way to check if that tests still pass. Then the other `getFilePath` methods that call this could be inlined and made more simple. Like

```java
public static File getFilePath(File parent, File path) {
    return getFilePath(parent, path.getPath());
}
```

would be rewritten to

```java
public static File getFilePath(File parent, File path) {
    return parent.toPath().resolve(path.toPath()).toFile();
}
```